[stable/fluent-bit] Upgrade to Fluent Bit v1.1.0

edsiper opened the pull request:
#### What this PR does / why we need it:

Upgrade to the latest stable version of Fluent Bit v1.1.0.

#### Special notes for your reviewer:

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

edsiper committed:
[stable/fluent-bit] Upgrade to Fluent Bit v1.1.0

Signed-off-by: Eduardo Silva <eduardo@treasure-data.com>

mtparet commented on stable/fluent-bit/values.yaml:
```diff
@@ -5,8 +5,8 @@ on_minikube: false
-    tag: 1.0.6
-  pullPolicy: IfNotPresent
+    tag: 1.1.0
+  pullPolicy: Always
```
Is the always policy needed by default ?

edsiper replied:
Ideally yes (maintainers stats), this was changed by mistake in a previous version, user can always adjust it as required .

mtparet replied:
Ok, thanks for the answer!

edsiper committed:
Merge branch 'master' into flb-1.1.0

edsiper committed:
Merge branch 'master' into flb-1.1.0

kfox1111 commented on stable/fluent-bit/values.yaml:
```diff
@@ -5,8 +5,8 @@ on_minikube: false
-    tag: 1.0.6
-  pullPolicy: IfNotPresent
+    tag: 1.1.0
+  pullPolicy: Always
```
Why change the pull policy? The image shouldn't ever change.

edsiper replied:
I am reverting to the original value we had, Pull Policy was changed by 00a84538d7 in January this year without noticing it. 

From a maintainer perspective that's' our only metric to get a notion of the number of deployments. Users can still change to their preferred value.

kfox1111 replied:
hmm. ok. Makes sense.

edsiper replied:
@kfox1111 can you help to review the new property of Kube_Tag_Prefix ?

kfox1111 replied:
hmm... so there is:
 Match               {{ .Values.filter.kubeTag }}.*
Kube_Tag_Prefix     {{ .Values.filter.kubeTag }}.

Both set to the same values, with one with a star and the other with just a dot. is that right? Do we need them to be separably set?

kfox1111 replied:
kubeTag is set by default to 'kube'. but the docs for Kube_Tag_prefix show something like: kube.var.log.containers.

Maybe kubeTagPrefix should be its own value so its independently set able.

edsiper replied:
actually Kube_Tag_Prefix by default must point to: 

```
{{ .Values.filter.kubeTag }}.var.log.containers.
```
here is an explanation of the changes and config to consider:

https://docs.fluentbit.io/manual/installation/upgrade_notes

edsiper replied:
hmm make sense, I will do the change

edsiper replied:
updated here: https://github.com/helm/charts/pull/13702/commits/2d7e21935516f8376c5f34dc2dd4ebc796f4c0c6

edsiper replied:
@kfox1111 can you test it ?

kfox1111 replied:
I don't have a quick place to do a test at the moment. sorry.

edsiper committed:
[stable/fluent-bit] add new filter.KubeTagPrefix option

Starting from Fluent Bit 1.1.0, a new option called Kube_Tag_Prefix has
been introduced to help to discover Kubernetes metadata coming from
monitored files. This change adjust to the upgrade requirements on the
following document:

https://docs.fluentbit.io/manual/installation/upgrade_notes

Signed-off-by: Eduardo Silva <eduardo@treasure-data.com>
